web/src/compose_setup.js

```diff
@@ -178,9 +179,15 @@ export function initialize() {
                     // user-triggered edit to that field.
                     $input.trigger("input");
 
-                    // TODO: Probably this should also renarrow to the
-                    // new topic, if we were currently viewing the old
-                    // topic, just as if a message edit had occurred.
+                    // Renarrow to the unresolved topic if currently viewing the resolved topic.
+                    const current_filter = narrow_state.filter();
+                    const channel_name = sub_store.maybe_get_stream_name(stream_id);
+                    if (current_filter && current_filter.has_topic(channel_name, topic_name)) {
```

Comment:
Why is the check `current_filter.has_topic`? I would have thought we'd be trying to check if we're exactly in that conversation view.

Reply:
OK so that check is whether it has the channel/topic operators; but not whether we're in a search view with additional operators. Probably checking `(filter.is_conversation_view() || filter.is_conversation_view_with_near()) && current_filter.has_topic()` would be a precise check.
```
    has_topic(stream_name: string, topic: string): boolean {                                            
        return this.has_operand("channel", stream_name) && this.has_operand("topic", topic);            
    }                                                                                                   
```

Reply:
I don't understand why we'd need to know anything more than whether we have the relevant channel/topic operators, since this is a very specific case - where there's no message in the topic and the topic is resolved

Reply:
While it's quite unlikely to ever come up, I think it'd still be incorrect to navigate you from a `stream:foo topic:"{resolved}bar" sender:me` to a stream/topic view in this situation. I'll add the check I suggested.